content/en/hosting/cht/migration/preparing-for-5.md

```diff
@@ -0,0 +1,161 @@
+
+To check if any of your users are impacted, check the [user-devices API](/building/reference/api/#get-apiv2exportuser-devices). This is an authenticated API endpoint and will return JSON for all users for all time.  As this may include multiple entries per user, it's important to filter out duplicate users and of course filter out users who are running Chrome 107 or later. 
+
+Before starting, be sure `curl`, `jq` and `sort` are installed. In Ubuntu server, this looks like: `sudo apt update&&sudo apt install jq coreutils curl`. After confirming these tools are installed, save the output of the API to a JSON file being sure to replace `user`, `password` and `URL` with their correct values for your production instance:
+
+```shell
+curl https://user:password:URL/api/v2/export/user-devices > devices.json 
+```
+
+**Note**: for large instances with more than 2,000 users, run the API call after hours.  It can have an adverse impact on CHT server performance.
+
+Now that you have the `devices.json` JSON file, flatten it into a CSV file with just one row per user. Note that we're only finding users of the APK (`select(.apk|length> 0)`) and the Chrome browser (`select(.browser.name == "Chrome"`). This excludes desktop and Firefox users which will show up in the JSON as well.  Finally, we're also only showing users active since `2025-04-01`, 6 months ago as of this writing (`select(.date > "2025-04-01")`).  Be sure to update this date to be a relative 6 months ago from when you run the command:
+
+```shell
+jq -r '.[] |  select(.apk|length> 0) 
+  | select(.browser.name == "Chrome")
+  | select(.date > "2025-04-01")
+  | "\(.deviceId) \(.user) \(.date) \(.browser.version)"' \
+  devices.json | sort -u -k1,2r \
+  | sort -u -k1,1 | cut -f2,3,4 -d' '| sort -rh -k3 \
+  | sed 's/ /,/g' > devices.csv
+```
+
+The resulting file `devices.csv` will have all of your users sorted by the version of Chrome they use.  Scroll to the bottom and find any users running a version lower than `107` and ensure they update to a newer version.  The easiest way to upgrade is to open the Play Store and update any out of date software, being sure to update [WebView](https://play.google.com/store/apps/details?id=com.google.android.webview&hl=en).
+
+For instances with users who share devices, or have multiple devices they log in on, they may have multiple entries. Thi is because the CHT creates one entry per device per user. Be sure to upgrade all affected devices!
```

Comment:
```suggestion
For instances with users who share devices, or have multiple devices they log in on, they may have multiple entries. This is because the CHT creates one entry per device per user. Be sure to upgrade all affected devices!
```